### PR TITLE
Tolerate some errors in VM Create()

### PR DESCRIPTION
**Pull request by maya-r**
We add the permission in BeforeEach and immediately try to use
this permission which might be too quick.
This solution has been adopted in createVmSuccess in
tests/storage/datavolume.go, copy it.

**What this PR does / why we need it**:
Fixes flakes seen, for example [here](https://prow.ci.kubevirt.io/view/gcs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.24-sig-storage/1552448762796838912)

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
I think it would be better if we could verify permissions just after creating them in `BeforeEach`, but this PR solves the problem and is good for tests.

**Comment by maiqueb** [comment]
This does look weird to me, but unfortunately I am not storage-savy. 

I'm removing myself from the reviewers list. 

/uncc

**Comment by akalenyu** [comment]
I think we're fine with this; I am not sure there are better ways to poll on RBAC being applied (if at all)
We've seen the same in CDI in the past:
https://github.com/kubevirt/containerized-data-importer/pull/1971
https://github.com/kubevirt/containerized-data-importer/pull/882

/lgtm

File: tests/storage/restore.go [hunk 1]
```diff
@@ -116,9 +116,12 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 		vm.Spec.Running = &t
 		var gracePeriod int64 = 10
 		vm.Spec.Template.Spec.TerminationGracePeriodSeconds = &gracePeriod
-		vm, err := virtClient.VirtualMachine(vm.Namespace).Create(vm)
-		Expect(err).ToNot(HaveOccurred())
 		Eventually(func() bool {
+			// sometimes it takes a bit for permission to actually be applied so eventually
+			vm, err := virtClient.VirtualMachine(vm.Namespace).Create(vm)
+			if err != nil {
+				return false
```
Comment (mhenriks) [inline tests/storage/restore.go:123]: Should tolerate "already exists" here for the case when create succeeds and the the following get fails

**Comment by maya-r** [comment]
mhenriks made some good points in a private chat about how this code will fail if the create already succeeded but other parts failed, and we try to re-create again (fails with already exists). It is also going to return an outdated vmi due to scoping issues.

The new version fixes both these problems and uses the same eventually loop that the code in tests/storage/datavolume.go does.

**Comment by maya-r** [comment]
/hold
I see that the same code failed at the introduced code. It might be a different problem

**Comment by maya-r** [comment]
okay, the problem is that we update the vm object with the failed attempt :)

**Comment by akalenyu** [comment]
> okay, the problem is that we update the vm object with the failed attempt :)

so not related to this change? where does it happen?

**Comment by maya-r** [comment]
/hold cancel

the problem was we did `vm, err := create(vm)` in a loop until it succeeds
If this fails, running it a second time will keep failing - vm is now an empty object.

**Comment by brybacki** [comment]
/lgtm
I think the solution is good enough for this test case

**Comment by xpivarc** [comment]
I think it would make more sense to use auth API to verify your user(SA) can do what you want. The proposed solution affects all the tests which I dislike and might cause problems in the future. Please reconsider the approach.

**Pull request merged**